### ISSUE TITLE
Have library power to SPDIF out?

### ISSUE THREAD

**Issue by mmar22** [schreibfaul1/ESP32-audioI2S#616]
What you mean can I2S to SPDIF sw code be used here?

**Comment by schreibfaul1** [comment]
No, I don't have the possibility to test this either.
Have a look here: https://github.com/earlephilhower/ESP8266Audio/blob/master/src/AudioOutputSPDIF.cpp
Maybe you can realise this yourself.

**Comment by mmar22** [comment]
Ofcourse i dont ask your code, but your meaning. Too question is how is max samplerates for every your codecs...

**Comment by schreibfaul1** [comment]
The samplerates depends on the source, max is 48KHz, bitrate max 320Kbit/s

**Comment by mmar22** [comment]
This is clean, but exist hires radio 96kHz/24bit 1300kbit i ask if you have tested ESP32 limits?

**Comment by schreibfaul1** [comment]
No, we are not travelling in the Hi End range with this hardware, max is 16 bit/sample

**Comment by mmar22** [comment]
I use spdif code on other project, but here for speedup test can you help where in your code i can set data buffer callback for fill spdif instead iľs out.

**Comment by schreibfaul1** [comment]
There are two events that can be used to decouple the I2S data:
`void audio_process_extern(int16_t* buff, uint16_t len, bool *continueI2S)`
`void audio_process_i2s(uint32_t* sample, bool *continueI2S)`
The latter is used by the [ESPuino](https://github.com/biologist79/ESPuino) project for Bluetooth a2dp source

**Comment by mmar22** [comment]
I implement SPDIF and test it now, but i use too lvgl and when i place in loop
`audio.loop();
  lv_timer_handler(); /* let the GUI do its work */
  delay( 5 );`
then i have issue on any animation audio is corrupted.
I mean require dedicated task for audio.loop() , but how stack and other params is optimal ? 
Can you recommend any

**Comment by mmar22** [comment]
@schreibfaul1  i test more and corruption in audio is on AAC station. ANy tip? I use task on core 0 now
`void task_AAloop(void)
{
  for( ;; )
  {
    audio.loop();
    vTaskDelay(5 / portTICK_PERIOD_MS);
  }
}
`
